src/snowflake/snowpark/mock/_functions.py

```diff
@@ -929,3 +930,99 @@ def mock_to_variant(expr: ColumnEmulator):
     res = expr.copy()
     res.sf_type = ColumnType(VariantType(), expr.sf_type.nullable)
     return res
+
+
+DATETIME_PART_TO_ALIASES = {
+    "year": {"y", "yy", "yyy", "yyyy", "yr", "years", "yrs"},
+    "quarter": {"q", "qtr", "qtrs", "quarters"},
+    "month": {"mm", "mon", "mons", "months"},
+    "week": {"w", "wk", "weekofyear", "woy", "wy"},
+    "day": {"d", "dd", "days", "dayofmonth"},
+    "hour": {"h", "hh", "hr", "hours", "hrs"},
+    "minute": {"m", "mi", "min", "minutes", "mins"},
+    "second": {"s", "sec", "seconds", "secs"},
+    "millisecond": {"ms", "msec", "milliseconds"},
+    "microsecond": {"us", "usec", "microseconds"},
+    "nanosecond": {
+        "ns",
+        "nsec",
+        "nanosec",
+        "nsecond",
+        "nanoseconds",
+        "nanosecs",
+        "nseconds",
+    },
+}
+
+DATETIME_PARTS = set(DATETIME_PART_TO_ALIASES.keys())
+ALIASES_TO_DATETIME_PART = {
+    v: k for k, l in DATETIME_PART_TO_ALIASES.items() for v in l
+}
+DATETIME_ALIASES = set(ALIASES_TO_DATETIME_PART.keys())
+
+
+def unalias_datetime_part(part):
+    if part in DATETIME_PARTS:
+        return part
+    elif part in DATETIME_ALIASES:
+        return ALIASES_TO_DATETIME_PART[part]
+    else:
+        raise ValueError(f"{part} is not a recognized date or time part.")
+
+
+def cast_to_datetime(date):
+    return datetime.datetime.fromordinal(date.toordinal())
+
+
+def add_years(date, duration):
+    return date.replace(year=date.year + duration)
+
+
+def add_months(scalar, date, duration):
+    # Keep months in 1..12
+    month = (date.month - 1 + scalar * duration) % 12 + 1
+
+    # Add years as needed
+    year = date.year + (date.month - 1 + scalar * duration) // 12
+
+    return date.replace(year=year, month=month)
+
+
+def add_timedelta(unit, date, duration, scalar=1):
+    return date + datetime.timedelta(**{f"{unit}s": duration * scalar})
+
+
+@patch("dateadd")
+def mock_dateadd(part, expr1, expr2):
+    # Extract a standardized name
+    part = unalias_datetime_part(part)
+    sf_type = expr1.sf_type
+    ts_type = ColumnType(TimestampType(TimestampTimeZone.NTZ), expr1.sf_type.nullable)
```

Comment:
curious about why TimestampTimeZone.NTZ here, not LTZ/TZ

Reply:
TimeZone.NTZ is the type behavior I observed when running the equivalent sql statements directly.